Migrate the KeyDistribution_BB84 kata (#1624)
microsoft/qsharp: katas/content/key_distribution/index.md

# Quantum Key Distribution

@[section]({
    "id": "key_distribution__overview",
    "title": "Overview"
})

> Quantum key distribution is a type of quantum communication protocols that allow two parties to generate shared secret keys - random strings of bits known only to those two parties. These shared keys can then be used for a variety of different classical cryptographic protocols like encryption or authentication.

Reviewer: ```suggestion
Quantum key distribution is a type of quantum communication protocol that allow two parties to generate shared secret keys - random strings of bits known only to those two parties. These shared keys can then be used for a variety of different classical cryptographic protocols like encryption or authentication.
```

Author: ```suggestion
Quantum key distribution is a type of quantum communication protocol that allows two parties to generate shared secret keys - random strings of bits known only to those two parties. These shared keys can then be used for a variety of different classical cryptographic protocols like encryption or authentication.
```

Author: Good point, but then we need to change the verb too

In the first basis, the computational basis, Alice prepares the states $\ket{0}$ and $\ket{1}$ where $\ket{0}$ represents the key bit value `0` and $\ket{1}$ represents the key bit value `1`.
The second basis, Hadamard basis (sometimes also called the diagonal basis), uses the states $\ket{+} = \frac{1}{\sqrt2}(\ket{0} + \ket{1})$ to represent the key bit value `0`, and $\ket{-} = \frac{1}{\sqrt2}(\ket{0} - \ket{1})$ to represent the key bit value `1`.

<table>
  <tr>
    <th style="text-align:center">Basis</th>
    <th style="text-align:center">Bit 0</th>
    <th style="text-align:center">Bit 1</th>    
  </tr>
  <tr>
    <td style="text-align:center">Computational basis</td>
    <td style="text-align:center">$\ket{0}$</td>
    <td style="text-align:center">$\ket{1}$</td>
  </tr>
  <tr>
    <td style="text-align:center">Hadamard basis</td>
    <td style="text-align:center">$\ket{+}$</td>
    <td style="text-align:center">$\ket{-}$</td>
  </tr>
</table>

The bases used in the protocol are selected such that if an eavesdropper tries to measure a qubit in transit and chooses the wrong basis, then they just get a 0 or 1 measurement result with equal probability.

> Alice has to make two random choices per qubit she prepares, one for which basis to prepare in, and the other for what bit value she wants to send, choosing each option with $50\%$ probability.

Reviewer: ```suggestion
Alice has to make two random choices for each qubit she prepares, one for which basis to prepare in, and the other for what bit value she wants to send, choosing each option with $50\%$ probability.
```
